### DISCUSSION TITLE
Consider adding `Week` option to PeriodType enum

### DISCUSSION THREAD

**Discussion by JSv4** [Open-Cap-Table-Coalition/Open-Cap-Format-OCF#146] [Suggestions] [answered]
**Per issue #56, importing comments / suggestions to our static Google Doc into GitHub for discussion**

@ashawber proposed adding a `WEEK` option to our PeriodType enum:

https://github.com/Open-Cap-Table-Coalition/Open-Cap-Format-OCF/blob/9edd343c01303f82a2b3e5161faebab7637e27ad/schema/enums/PeriodType.schema.json#L1-L8

Seems sensible to me. I'd just want us to think through how to handle partial weeks. If you have a period of X weeks from Y date, which is in the middle of the week, do you just count off periods of 7 days from date Y (in which case, could you just set a period of some multiple of 7 days)? 

If we're not just counting 7 days from an arbitrary date, you start to run into calendar quirks. In the example above, do you count that initial partial week as a whole week? Does the period end in the middle of the final week? At the end of it?

There may be some other corner cases here that will not be straight-forward, which could cause compatibility problems. Unless we have examples of situations in which it's not possible to express a period in days, months or years, perhaps we don't need this?

**Comment by ashawber** [comment]
Weekly vesting is more common in equity grants to consultants/contractors.

Good point re how to handle partial weeks, but don't we have a similar problem with partial months?

A typical monthly vesting schedule would be something like this: "Prior to your Termination of Service, 1/48 of the Shares will vest each month on the same day of the month as the Vesting Commencement Date (and if there is no corresponding day, the last day of the month)." 

But not all vesting schedules are that precise.

Maybe the solution is to make the system smart enough for the user to define the rules for how weekly/monthly vesting works. E.g. Is it the same day of the week/month, or each 7/30 day period? We give them some options so they can align the system's calculation with the underlying paperwork.

**Comment by JSv4** [comment, marked as answer]
@ashawber, @pjohnmeyer is working on a substantial overhaul of how the vesting works, and this should be addressed by that change. Please stay tuned for that pull request. Once it's in the repo, would be great to get your take on whether we're still missing the mark here. Thanks for all of your comments, BTW. They've been really helpful and led to a number of good changes.
